### PR TITLE
Fix CI warnings

### PR DESCRIPTION
**Pull request by feliwir**
Need to use the latest checkout action version, since old NodeJS actions will be removed

### CODE REVIEW

**Comment by feliwir** [comment]
See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Review by OmniBlade — COMMENTED** [review]
This looks fine to me, but given the time that has passed I'd like to rebase it on the latest develop and retest to make sure there is nothing else that needs a version bump.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
         os: [ubuntu-20.04, ubuntu-22.04]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (janisozaur) [inline .github/workflows/linux.yml]: You need v4 by now.

```suggestion
      - uses: actions/checkout@v4
```

https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
         platform: [x86, amd64]
           
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
```
Comment (janisozaur) [inline .github/workflows/windows.yml]:
```suggestion
    - uses: actions/checkout@v4
```

**Review by janisozaur — CHANGES_REQUESTED** [review]

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**